plugins/outputs/signalfx/signalfx.go

```diff
@@ -301,6 +302,7 @@ func (s *SignalFx) emitDatapoints() {
 	for {
 		select {
 		case <-s.done:
+			s.wg.Done()
```

Comment:
It seems clearer to put the wait group release logic in an anonymous function that calls this function above in the `Connect` method where you are incrementing the wait group.

```
go func() {
  s.emitDatapoints()
  s.wg.Done()
}()
```

Reply:
Also you should do the `wg.Add(2)` *before* starting these goroutines because if the count goes < 0 the whole thing blows up.